backend/Events/NightTimeVoteEvents.js

```diff
@@ -0,0 +1,54 @@
+const RoleEnum = require('../domain/Enum/Role');
+const SuspectRevealDTO = require('../domain/DTO/response/SuspectRevealDTO');
+
+/**
+ * Event handlers and logic for `mafia-vote`, 'medic-vote' and `detective-vote`
+ * This involves the functionality for night-time voting, which includes:
+ * - Mafia choosing who to kill before the next round
+ * - Medics choosing who to save
+ * - Detectives choosing who to suspect
+ * @param {any} io
+ * @param {any} socket
+ * @param {MafiaGame} mafiaGame
+ */
+exports.loadNightTimeEvents = (io, socket, mafiaGame) => {
+    /**
+     * Handler for 'mafia-vote', sets the chosen player for the Mafia to the player specified in the message.
+     * In future iterations, this will be changed to tally votes from mafia members before setting the chosen player.
+     */
+    socket.on('mafia-vote', (mafiaVoteObj) => {
+        let room = socket.player.roomID;
+        room.voteHandler.addMafiaVote(
```

Comment:
I may be wrong but socket.player.roomID gives the roomID which is a string. I believe you want to go let room = mafiaGame.gameRoomsDict[socket.player.roomID] to get the room object. then you can invoke the methods.

Reply:
Fixed in recent commit